tensorflow/java/src/main/java/org/tensorflow/Tensor.java

```diff
@@ -317,7 +433,15 @@ private void throwExceptionIfTypeIsIncompatible(Object o) {
     }
   }
 
-  private static native long allocate(int dtype, long[] shape);
+  private static ByteBuffer getBuffer(long nativeHandle) {
+    return buffer(nativeHandle).order(ByteOrder.nativeOrder());
+  }
+
+  private static native ByteBuffer buffer(long handle);
+
+  private static native long allocate(int dtype, long[] shape, long sizeInBytes);
```

Comment:
I agree with your latter point.  If I recall correctly, it would require that the element counting code within the ND array codepath be moved from C to Java.  Since we have `flattenedNumElements` in the managed test code anyway, I have no objection.